src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs

```diff
@@ -13,30 +13,83 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 // </copyright>
+
 using System;
-using System.Collections.Generic;
 using System.IO;
 using System.Net;
 using System.Threading;
-using OpenTelemetry.Metrics.Implementation;
+using System.Threading.Tasks;
+using OpenTelemetry.Exporter.Prometheus.Implementation;
 
-namespace OpenTelemetry.Exporter.Prometheus.Implementation
+namespace OpenTelemetry.Exporter.Prometheus
 {
-    internal class MetricsHttpServer
+    /// <summary>
+    /// A HTTP listener used to expose Prometheus metrics.
+    /// </summary>
+    public class PrometheusExporterMetricsHttpServer
     {
-        internal List<Metric> Metrics;
-
+        private readonly PrometheusExporter exporter;
         private readonly CancellationToken token;
         private readonly HttpListener httpListener = new HttpListener();
+        private readonly object lck = new object();
+
+        private CancellationTokenSource tokenSource;
+        private Task workerThread;
 
-        public MetricsHttpServer(List<Metric> metrics, PrometheusExporterOptions options, CancellationToken token)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrometheusExporterMetricsHttpServer"/> class.
+        /// </summary>
+        /// <param name="exporter">The <see cref="PrometheusExporter"/> instance.</param>
+        /// <param name="options">The <see cref="PrometheusExporterOptions"/> instance.</param>
+        /// <param name="token">A <see cref="CancellationToken"/> that can be used to stop the Htto Server.</param>
+        public PrometheusExporterMetricsHttpServer(PrometheusExporter exporter, PrometheusExporterOptions options, CancellationToken token)
```

Comment:
`PrometheusExporter` was already created with the set of `PrometheusExporterOptions`. It's better to get config values from the exporter than pass them again here. This way the situation when one passed different set of properties to exporter and server will be eliminated